src/python/pants/option/subsystem.py

```diff
@@ -96,21 +97,32 @@ def get_scope_info(cls) -> ScopeInfo:
         return cls.create_scope_info(scope=cls.options_scope, subsystem_cls=cls)
 
     @classmethod
+    @deprecated(
+        removal_version="2.12.0.dev0",
```

Comment:
This needs input from several people, given that it is disruptive to plugin authors. I recommend either starting a Slack discussion or pinging multiple reviewers.

I generally am +1 for an early removal version like this, as it unblocks improvements to our underlying options code. It's not pretty, but plugins like Toolchain I believe could use conditional logic if they need to handle both APIs.